apps/prairielearn/src/models/course-instances.sql

```diff
@@ -22,7 +22,13 @@ SELECT
     OR ia.id IS NOT NULL
     OR cip.course_instance_role > 'None',
     FALSE
-  ) AS has_course_instance_permission_view
+  ) AS has_course_instance_permission_view,
+  COALESCE(
+    $is_administrator
+    OR ia.id IS NOT NULL
+    OR cip.course_instance_role = 'Student Data Editor',
```

Comment:
Nit, this will be safer if we ever add any higher-privilege roles.

```suggestion
    OR cip.course_instance_role >= 'Student Data Editor',
```

Reply:
Addressed in 69f0ba1